### DISCUSSION TITLE
Maven from Artifactory Virtual Repository

### DISCUSSION THREAD

**Discussion by EvertonSA** [renovatebot/renovate#18803] [Request Help] [answered]
hi colleagues, please correct me if i'm wrong, but if I understood correctly, renovate is not capable of using artifactory maven virtual repositories to fetch for newer dependencies if settings.xml does not exists on repository.

My settings.xml **is not present in the repository**, but injected in pipeline time. This works great in my case where I need to maintain 1 `settings.xml` for many different repositories. When trying renovatebot to update dependencies I'm always getting this error: `Dependency lookup unauthorized. Please add authentication with a hostRule ...` even though config.js has the correct token. 

I managed to get renovate to fetch from private registry by setting `registryUrls` to the maven virtual repo, and I confirmed in renovate debug logs that is trying to fetch from correct URL, but still fails with authentication even with `matchHost` + `token`. 

I understood from this issue that renovate does not work when settings.xml is not present, https://github.com/renovatebot/renovate/issues/13925, is my assumption correct?

![image](https://user-images.githubusercontent.com/30481051/200366591-caf3e80a-a611-43f2-9d21-b57ee094d64f.png)

content of config.js in repository were pom.xml exists:

```
module.exports = {
  hostRules: [
    {
      hostType: 'maven',
      matchHost: 'https://artifactory-something.com/artifactory/team-something-maven/',
      token: process.env.ARTIFACTORY_TOKEN
    },
  ],
  packageRules: [
    {
      matchDatasources: ['maven'],
      registryUrls: ['https://artifactory-something.com/artifactory/team-something-maven/'],
    },
  ],
};
```
logs:
![image](https://user-images.githubusercontent.com/30481051/200367978-823f101a-b1b7-4e6f-8e4d-636ee919be4c.png)

**Comment by Shegox** [comment]
Not sure if that helps you, but we use following packageRules/hostRules for lookup with JFrog Artifactory. Notice that we use `authType: Basic` and username and pasword (the API key as password). Additionally we only use `/build.releases` directly without the additional `/artifactory/` in the URL path.

```json
{
  "hostRules": [
    {
      "hostType": "maven",
      "matchHost": "https://artifactory-something.com/build.releases",
      "username": "$ARTIFACTORY_USER",
      "password": "$ARTIFACTORY_API_KEY",
      "authType": "Basic"
    }
  ],
  "packageRules": [
    {
      "matchManagers": [
        "maven"
      ],
      "registryUrls": [
        "https://artifactory-something.com/build.releases"
      ]
    }
  ]
}        
```

**Reply by EvertonSA** [reply]
thanks for the input, do you have a settings.xml on your pom repositories?

**Comment by EvertonSA** [comment]
@Chumper I know it sucks being tagged by maven newbie users, but any help is appreciated. Do you know is not having settings.xml for can be an issue? My usecase is, no matter what, we should always fetch from artifactory, never from internet. The global settings.xml I have in pipeline time does that and it uses a custom id instead of `central`.

**Reply by Chumper** [reply]
Afaik you will not be able to get of maven as default mirror.
I had a few PRs ready but they were not fleshed out enough to solve this problem.

**Comment by EvertonSA** [comment, marked as answer]
as for reference, at the current state of this project, you will not be able to use renovate to pull pom dependencies from artifactory if your settings is external to the repo. meaning air gapped environments have to rely on proxy or change settings.xml strategy.
